lib/elixir/lib/module/checker.ex

```diff
@@ -3,58 +3,97 @@ defmodule Module.Checker do
 
   @moduledoc false
 
-  def verify(module_map, binary, cache) do
-    warnings = warnings(module_map, cache)
-    binary = chunk(binary, module_map)
-    {binary, warnings}
+  def verify(module, cache) do
+    case prepare_module(module) do
+      {:ok, map} -> {build_chunk(map), warnings(map, cache)}
+      :error -> {nil, []}
+    end
+  end
+
+  defp prepare_module({_module, map}) when is_map(map) do
+    {:ok, map}
+  end
+
+  defp prepare_module({module, binary}) when is_binary(binary) do
+    with {:ok, debug_info} <- debug_info(module, binary),
+         {:ok, checker_info} <- checker_chunk(binary) do
+      {:ok,
+       %{
+         module: module,
+         definitions: debug_info.definitions,
+         file: debug_info.file,
+         deprecated: checker_info.deprecated,
+         no_warn_undefined: checker_info.no_warn_undefined
+       }}
+    end
+  end
+
+  defp get_chunk(binary, chunk_name) do
```

Comment:
I think we could abstract this common logic into `:elixir_utils` (or maybe something in `Code`).

Reply:
Thanks, I forgot about this. I inlined it instead.